xarray/tests/datatree/test_treenode.py

```diff
@@ -370,6 +383,9 @@ def test_render_nodetree(self):
             "NamedNode('Ben')",
             "NamedNode('Kate')",
         ]
+
```

Comment:
@TomNicholas This test was passing, but probably incorrectly.  What should the `__str__` representation of a Named node look like?

From the test I was guessing:

``` 
NamedNode()
    NamedNode('Mary')
        NamedNode('Sam')
        NamedNode('Ben')
    NamedNode('Kate')
```

or as a string

``` 
"NamedNode()\n\tNamedNode('Mary')\n\t\tNamedNode('Sam')\n\t\tNamedNode('Ben')\n\tNamedNode('Kate')"
```

Should this be the `__str__` or the `__repr__`?
the `__repr__` is  currently:

```
"TreeNode(children={'Mary': TreeNode(children={'Sam': TreeNode(children={}), 'Ben': TreeNode(children={})}), 'Kate': TreeNode(children={})})"
```

Reply:
Ultimately both of these get overridden in `datatree.py` (`DataTree` is a subclass of `NamedNode`). So the printable representation of these internal classes should just be whatever is most convenient for us when debugging the code in the `treenode.py` module.

Reply:
I added a `__repr__` even if it's only ever going to be used in the test. Let me know if you hate it. 
7f2a178

Reply:
> I added a __repr__ even if it's only ever going to be used in the test. Let me know if you hate it.

That seems fine. I would probably keep the quotation marks though because `Sam` is never an object, always a string.

> And I've found the render module and am less sure what this should have been testing.

Yeah... Do we even use the render module anymore?? The whole thing might just be a vestigial appendage that originated from the first prototype being based on the [`anytree` library](https://github.com/c0fec0de/anytree).

Reply:
> Do we even use the render module anymore?? 
At least some of it is used in the `render` function and both `__repr__` and `__str__` for a datatree.

Reply:
I'll call it add quotation marks. 548536b00